src/app/containers/LoginPage/index.js

```diff
@@ -4,123 +4,124 @@ import React from 'react';
+          // eslint-disable-next-line react/prop-types
+          setRedirectState(true);
+        }
       } else {
-        this.context.setUserInfo({ ...this.context.userInfo, token: data.token });
-        // eslint-disable-next-line react/prop-types
-        this.props.history.push('/');
+        const { apiData } = await apiLogin({ username, password });
+        console.log(apiData);
+        if (!apiData || !apiData.token) {
+          setPopupState(true);
+          setApiProgress(API_PROGRESS.FAILED);
+        } else {
+          userInfo.setUserInfo({ ...userInfo.userInfo, userInfo: apiData.token });
+          // eslint-disable-next-line react/prop-types
+          setRedirectState(true);
+        }
       }
-    }
-  }
+    },
+    [data],
+  );
+
+  const handleClosePopup = React.useCallback(() => {
+    setPopupState(false);
+  });
 
-  handleChange(event) {
-    const { name, value } = event.target;
-    this.setState({
-      [name]: value,
-    });
-  }
+  const onButtonClick = React.useCallback(() => {
+    setRedirectState(true);
+  });
 
-  render() {
-    const { username, password } = this.state;
-    return (
-      <div className={styles.container}>
-        <div className={styles.formContainer}>
-          <form onSubmit={this.handleSubmit}>
-            <h5 className={styles.titleContent}>Đăng nhập:</h5>
-            <InputText
-              label="Tên đăng nhập"
-              divStyle={styles.username}
-              labelStyle={styles.usernameLabel}
-              inputStyle={styles.usernameInput}
-              type="text"
-              name="username"
-              value={username}
-              onChange={this.handleChange}
-            />
-            <InputText
-              label="Mật khẩu"
-              divStyle={styles.password}
-              labelStyle={styles.passwordLabel}
-              inputStyle={styles.passwordInp}
-              type="password"
-              name="password"
-              value={password}
-              onChange={this.handleChange}
-            />
-            <button
-              className={styles.loginBtn}
-              type="submit"
-              disabled={this.state.apiProgress === API_PROGRESS.REQ}
-            >
-              Đăng nhập
-            </button>
-          </form>
-        </div>
-        {this.state.showFalsePopup && (
-          <>
-            <div className={styles.overlay} />
-            <PopupFailed closePopup={this.closePopup} />
-          </>
-        )}
-      </div>
-    );
-  }
+  return (
+    <MainPanel.Container>
+      {redirectState ? <Redirect to="/" /> : null}
+      {popupState ? (
+        <Popup
+          onClose={handleClosePopup}
+          title="Đăng nhập không thành công"
+          content="Tên đăng nhập hoặc mật khẩu không chính xác. Vui lòng thử lại."
+          buttonText="OK"
+          variant="error"
+          onButtonClick={onButtonClick}
+        />
+      ) : (
+        ''
+      )}
```

Comment:
can use `&&` operation to shorten the logic

```suggestion
      {popupState && (
        <Popup
          onClose={handleClosePopup}
          title="Đăng nhập không thành công"
          content="Tên đăng nhập hoặc mật khẩu không chính xác. Vui lòng thử lại."
          buttonText="OK"
          variant="error"
          onButtonClick={onButtonClick}
        />
      )}
```